### PR TITLE
Service Accounts - add rest specs and tests

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds rest spec files and associated tests for APIs of the new service accounts feature.

### CODE REVIEW

**Comment by ywangd** [comment]
@elastic/clients-team These new APIs will be released as beta in 7.13. This PR adds the rest spec files and relevant tests. Changes to HLRC are planned for 7.14. Thanks!

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.get_service_tokens.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "security.get_service_tokens":{
```
Comment (tvernum) [inline rest-api-spec/src/main/resources/rest-api-spec/api/security.get_service_tokens.json:2]: I think this one is mismatched - The endpoint is for `/credential/` but the API is specifically named about tokens. I think it should be named  `get_service_credentials` instead.

File: x-pack/plugin/build.gradle [hunk 2]
```diff
@@ -47,25 +47,30 @@ tasks.named("testJar").configure {
 }
 
 // location for keys and certificates
-File keystoreDir = file("$buildDir/keystore")
-File nodeKey = file("$keystoreDir/testnode.pem")
-File nodeCert = file("$keystoreDir/testnode.crt")
+File extraResourceDir = file("$buildDir/keystore")
```
Comment (tvernum) [inline x-pack/plugin/build.gradle:50]: I don't get this.  
Why is the directory named `keystore` if it contains general resources?

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+---
+setup:
+  - skip:
+      features: headers
```
Comment (tvernum) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml:4]: Do we need to skip on the `headers` feature? We don't seem to use it.

File: x-pack/plugin/build.gradle (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/build.gradle:50]: Good catch! It's a leftover piece. Update to `extra_resource`.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml (hunk 3 above)
Comment (ywangd) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/service_accounts/10_basic.yml:4]: You are right. The copy/paste behaviour is showing through 🤦

File: rest-api-spec/src/main/resources/rest-api-spec/api/security.get_service_tokens.json (hunk 1 above)
Comment (ywangd) [reply rest-api-spec/src/main/resources/rest-api-spec/api/security.get_service_tokens.json]: I changed it to `get_service_credentials`. Should the Java code also be updated for consistency and correctness, e.g. rename `GetServiceAccountTokensAction` to `GetServiceAccountCredentialsAction`? I can raise a separate PR just for renames and it should be trivial.

Comment (ywangd) [reply rest-api-spec/src/main/resources/rest-api-spec/api/security.get_service_tokens.json]: I raised https://github.com/elastic/elasticsearch/pull/71917 for it.
It has renames discussed here and also another rename that I wanted to have for a while.

**Review by tvernum — APPROVED** [review]
LGTM

**Pull request merged**